module/VuFind/src/VuFind/Role/PermissionProvider/SecureCookie.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Secure cookie permission provider.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Catalyst IT 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://www.vufind.org  Main Page
+ */
+
+namespace VuFind\Role\PermissionProvider;
+
+use Laminas\Session\Container;
+use VuFind\Cookie\CookieManager;
+
+/**
+ * Secure cookie permission provider.
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+class SecureCookie implements PermissionProviderInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Session container
+     *
+     * @var Container
+     */
+    protected $session;
+
+    /**
+     * Constructor
+     *
+     * @param CookieManager $cookieManager Cookie manager
+     * @param $session       Session container
+     */
+    public function __construct(
+        protected CookieManager $cookieManager,
+        Container $session
+    ) {
+        $this->session = $session;
+    }
+
+    /**
+     * Return an array of roles which may be granted the permission based on
+     * the options.
+     *
+     * @param mixed $options Options (cookieNames) provided from configuration.
+     *
+     * @return array
+     */
+    public function getPermissions($options)
```

Comment:
If you want this to work similarly to InsecureCookie, then `$options` should be a string or array of cookie names. Right now this method seems to be hard-coded to use a cookie named uuid, but I think you should actually be checking based on the $options value(s).

Also note that we should add an entry to permissions.ini describing the new provider so the documentation is complete.

Reply:
thanks @demiankatz I have made those changes to the getPermissions() function in SecureCookie.php

I've added added an entry to permissions.ini documenting this new secure cookie provider.

Please let me know if you think the documented description needs clarifying/changing?